src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RuntimeMetrics.cs

```diff
@@ -0,0 +1,244 @@
+
+        // Timer Metrics
+
+        private static readonly ObservableUpDownCounter<long> s_timerCount = s_meter.CreateObservableUpDownCounter(
+            "dotnet.timer.count",
+            () => Timer.ActiveCount,
+            unit: "{timer}",
+            description: "The number of timer instances that are currently active. An active timer is registered to tick at some point in the future and has not yet been canceled.");
+#endif
+
+        private static readonly ObservableUpDownCounter<long> s_assemblyCount = s_meter.CreateObservableUpDownCounter(
+            "dotnet.assemblies.count",
+            () => (long)AppDomain.CurrentDomain.GetAssemblies().Length,
+            unit: "{assembly}",
+            description: "The number of .NET assemblies that are currently loaded.");
+
+        private static readonly Counter<long> s_exceptionCount = s_meter.CreateCounter<long>(
+            "dotnet.exceptions.count",
+            unit: "{exception}",
+            description: "The number of exceptions that have been thrown in managed code.");
+
+        // CPU Metrics
+
+        private static readonly ObservableUpDownCounter<long> s_cpuCount = s_meter.CreateObservableUpDownCounter(
+            "dotnet.cpu.count",
+            () => (long)Environment.ProcessorCount,
+            unit: "{cpu}",
+            description: "The number of processors available to the process.");
+
+        private static readonly ObservableCounter<double> s_cpuTime = s_meter.CreateObservableCounter<double>(
+            "dotnet.cpu.time",
+            GetCpuTime,
+            unit: "s",
+            description: "The number of work items that are currently queued to be processed by the thread pool.");
+
+        public static bool IsEnabled()
+        {
+            return s_gcCollectionsCounter.Enabled
+                || s_gcObjectsSize.Enabled
+#if !NETFRAMEWORK && !NETSTANDARD
+                || s_gcMemoryTotalAllocated.Enabled
+                || s_gcMemoryCommited.Enabled
+                || s_gcHeapSize.Enabled
+                || s_gcHeapFragmentation.Enabled
+                || s_gcPauseTime.Enabled
+                || s_jitCompiledSize.Enabled
+                || s_jitCompiledMethodCount.Enabled
+                || s_jitCompilationTime.Enabled
+                || s_monitorLockContention.Enabled
+                || s_timerCount.Enabled
+                || s_threadPoolThreadCount.Enabled
+                || s_threadPoolCompletedWorkItems.Enabled
+                || s_threadPoolQueueLength.Enabled
+#endif
+                || s_assemblyCount.Enabled
+                || s_exceptionCount.Enabled
+                || s_cpuCount.Enabled
+                || s_cpuTime.Enabled;
+        }
+
+        private static IEnumerable<Measurement<long>> GetGarbageCollectionCounts()
+        {
+            long collectionsFromHigherGeneration = 0;
+
+            for (int gen = 2; gen >= 0; --gen)
```

Comment:
Yes, I suspect so. Will use `GC.MaxGeneration` as the starting value.

Reply:
That said, I'm unsure if a standalone GC would contribute to producing metrics. That's deeper than my knowledge of the GC bits go!

Reply:
Standalone GC plugs into all the same BCL GC APIs so I don't expect you will need to do anything special to accomodate it.

> AFAIK, it could happen if the user sets the [DOTNET_GCName environment variable](https://learn.microsoft.com/dotnet/core/runtime-config/garbage-collector#standalone-gc) to load a custom garbage collector.

As far as I know custom GCs for .NET are virtually non-existent. I'd be fine saying that these metrics are designed to work with a GC that is roughly similar to our existing GC. If anyone creates a GC that defines new generations these metrics would just be one of many scenarios that doesn't work correctly for them.